File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/ViewFileDetailsClickListener.java
package com.lasthopesoftware.bluewater.client.library.items.media.files;

import android.content.Intent;
import android.view.View;

import com.lasthopesoftware.bluewater.client.library.items.media.files.details.FileDetailsActivity;
import com.lasthopesoftware.bluewater.client.library.items.menu.NotifyOnFlipViewAnimator;
import com.lasthopesoftware.bluewater.client.library.items.menu.handlers.AbstractMenuClickHandler;

public class ViewFileDetailsClickListener extends AbstractMenuClickHandler {

	private final ServiceFile mServiceFile;
	
	public ViewFileDetailsClickListener(NotifyOnFlipViewAnimator viewFlipper, ServiceFile serviceFile) {
        super(viewFlipper);
		mServiceFile = serviceFile;
	}
	
	@Override
	public void onClick(View v) {
		Intent intent = new Intent(v.getContext(), FileDetailsActivity.class);
		intent.putExtra(FileDetailsActivity.FILE_KEY, mServiceFile.getKey());
		v.getContext().startActivity(intent);

        super.onClick(v);
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/files/details/ViewFileDetailsClickListener.kt
package com.lasthopesoftware.bluewater.client.browsing.files.details

import android.view.View
import com.lasthopesoftware.bluewater.client.browsing.files.ServiceFile
import com.lasthopesoftware.bluewater.client.browsing.files.details.FileDetailsActivity.Companion.launchFileDetailsActivity
import com.lasthopesoftware.bluewater.client.browsing.items.menu.NotifyOnFlipViewAnimator
import com.lasthopesoftware.bluewater.client.browsing.items.menu.handlers.AbstractMenuClickHandler
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.client.playback.file.PositionedFile

class ViewFileDetailsClickListener(
	viewFlipper: NotifyOnFlipViewAnimator,
	private val libraryId: LibraryId,
	private val positionedFile: PositionedFile,
	private val playlist: Collection<ServiceFile>,
) : AbstractMenuClickHandler(viewFlipper)
{
    override fun onClick(v: View) {
		v.context.launchFileDetailsActivity(libraryId, playlist, positionedFile.playlistPosition)
        super.onClick(v)
    }
}
